Newsletter Widget: improve the display of links.

jeherve opened the pull request:
Follow-up to #42571

## Proposed changes:

Let's improve the display of the Newsletter widget:

- External icons should have the proper margin on self-hosted, where we lack the icon styles outside of the editor. This brings in the necessary margin.
- Let's remove the dash in the widget description, and break the sentence into two.
- Let's avoid displaying the external link icon on the WordPress.com platform, where site owners are logged in to both wp-admin and WordPress.com (Calypso). They consequently do not need to log in when jumping from one environment to the other, and for them this is one single platform.

| **On self-hosted** | **On WordPress.com** |
|--------|--------|
| <img width="487" alt="Screenshot 2025-03-21 at 14 43 21" src="https://github.com/user-attachments/assets/84d76775-dd3a-4c55-9faf-5e95cc1a92d3" /> | <img width="477" alt="Screenshot 2025-03-21 at 14 43 16" src="https://github.com/user-attachments/assets/fd5c095e-4d53-4a90-99c2-64a91cbf6a09" /> | 

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

* See p1742544015819929-slack-C083ZPVVDTK

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

> [!NOTE]
> This is best tested on both WordPress.com and self-hosted.

1. Start with a site that's connected to WordPress.com, and where the Subscriptions feature is active.
2. Go to the main wp-admin dashboard.
3. Review the Newsletter dashboard widget.
    * Click on the links.
    * Ensure external links are displayed only when it makes sense.
    * Ensure the icons are displayed nicely, with enough margin.

jeherve committed:
Add changelog

jeherve committed:
Newsletter Widget: ensure external link icons have proper margin

The icon styles are not brought over to wp-admin yet, so we need to bring them with the Newsletter widget styles.

jeherve committed:
Adjust wording: remove dash and break into 2 sentences

jeherve committed:
Newsletter Widget: only display External Icon when necessary

On WordPress.com simple sites, folks are always connected to both wp-admin and WordPress.com (Calypso), so we do not have to show them an external link ; they will remain on the platform.

jeherve committed:
Remove unnecessary React import

jeherve commented on projects/plugins/jetpack/modules/subscriptions/newsletter-widget/src/style.scss:
```diff
@@ -7,123 +7,129 @@ $spacing-md: $grid-unit-15;      // 12px
+		text-decoration: none;
+	}
+
+	// External link icon needs a bit of spacing.
+	// This matches styles applied in the editor.
+	.components-external-link__icon {
+		margin-left: 0.5ch;
+	}
```
This is the only significant change here. The rest is my editor switching from spaces to tabs on save.

holdercp replied:
Thanks for the highlight. Should we be using Tabs in Jetpack? I was unaware when I wrote this. Something like an [EditorConfig](https://editorconfig.org/) might be helpful to enforce consistency. Or maybe I just don't have my editor set up correctly.

jeherve replied:
This is something that comes up from time to time. 

@Automattic/jetpack-garage What would you think about trying something like Stylelint (#11767) again, via `@wordpress/stylelint-config`?
https://github.com/WordPress/gutenberg/tree/trunk/packages/stylelint-config

tbradsha replied:
I can take a look at Stylelint.

anomiex replied:
The main reason we haven't done stylelint before is that when we've said "we can set it up, but someone else will have to make all the decisions about which rules to set", no one has stepped up to be the "someone else". 🙂

[@wordpress/stylelint-config](https://www.npmjs.com/package/@wordpress/stylelint-config) may be a good starting point, anyway.

holdercp replied:
I'm willing to spearhead setting up stylelint and making decisions on rules and such. I don't have really any history/context with rules that exist implicitly in the codebase but I can take a stab at it or help out in any way I can.

tbradsha replied:
We poked at this a bit Friday and will look deeper today; it seems Stylelint has some issues (e.g. scanning ignored files), but hopefully we can overcome them. :^)

holdercp reviewed (dismissed):
Looks great on Simple and links are correct.
<img width="552" alt="Screenshot 2025-03-21 at 9 33 33 AM" src="https://github.com/user-attachments/assets/9a0e4bdc-28fd-4a5e-862f-5b20e5bded43" />

Looks good on self-hosted:
<img width="525" alt="image" src="https://github.com/user-attachments/assets/290bbb13-44a8-46a9-9e80-1a0807eceb82" />

@jeherve I did want to bring up that both the "View subscriber stats" and "Newsletter settings" links load a Jetpack view within the dashboard instead of linking out to JPCloud.

Based on the criteria of, "As a customer if I click on a link and have to enter a set of login / password, I definitely consider it an external site." in the Slack thread, should these be external? On one hand I believe you need to login to connect JP to WPCOM so it fits, but on the other this experience feels weird to be marked as external:

<img width="1840" alt="Screenshot 2025-03-21 at 9 51 01 AM" src="https://github.com/user-attachments/assets/42f93991-4713-42a3-b25c-a0c02b84e9f7" />

Not a blocker IMO but I think it makes sense to keep the `wp-admin` based views not external.

jeherve committed:
Create new component to render link depending on the platform

jeherve committed:
Swich to using new component, and extract settings link

jeherve committed:
Add more tests

jeherve committed:
Bring in tracking for links, from #42660

Co-authored-by: Chris Holder <chris.holder@automattic.com>

jeherve committed:
Merge remote-tracking branch 'origin/trunk' into fix/newsletter-widget-external-links

jeherve committed:
Better wording for Dashboard link component prop

We do not necessarily always want to link out when the site is not WordPress.com.

jeherve committed:
Switch some of the links to always internal

jeherve committed:
Update tests to support tracking in extracted link component

jeherve commented:
@holdercp Sorry for the delay! I finally had some time to get back and address your feedback, as well as include the tracking you added to the widget a few days ago. It should be good for another review now.

holdercp commented on projects/plugins/jetpack/modules/subscriptions/newsletter-widget/src/helpers.ts:
```diff
@@ -1,8 +1,68 @@
+ * @param {string}  text     - The link text
+ * @returns {React.ReactElement} The link component
+ */
+export const DashboardLink = (
```
Love this! 💯

holdercp reviewed (dismissed):
Looks and works great. The only suggestion I have is to move the `<DashboardLink/>` component to `/components` as it's own component instead of in `helpers.ts` just to keep things tidy.

jeherve committed:
Merge remote-tracking branch 'origin/trunk' into fix/newsletter-widget-external-links

jeherve committed:
Extract new component to its own file, with own test

See https://github.com/Automattic/jetpack/pull/42645#pullrequestreview-2718661382

holdercp reviewed (approved):
Thanks for this work!
